js/apps/admin-ui/src/components/group/GroupPickerDialog.tsx

```diff
@@ -84,18 +84,25 @@ export const GroupPickerDialog = ({
             isSearching ? { search: filter } : null,
           ),
         );
-      } else if (!navigation.map(({ id }) => id).includes(groupId)) {
-        group = await adminClient.groups.findOne({ id: groupId });
-        if (!group) {
-          throw new Error(t("notFound"));
+      } else {
+        if (!navigation.map(({ id }) => id).includes(groupId)) {
+          group = await adminClient.groups.findOne({ id: groupId });
+          if (!group) {
+            throw new Error(t("common:notFound"));
+          }
         }
-        groups = group.subGroups!;
+        groups = await fetchAdminUI<GroupRepresentation[]>(
+          `groups/${groupId}/children`,
+          {
+            first: `${first}`,
+            max: `${max + 1}`,
+          },
+        );
       }
 
-      if (isSearching) {
-        count = (await adminClient.groups.count({ search: filter, top: true }))
-          .count;
-      }
+      count = (
+        await adminClient.groups.count({ search: filter, top: !groupId })
+      ).count;
 
       if (id) {
         existingUserGroups = await adminClient.users.listGroups({
```

Comment:
Looks like this request also doesn't have any dependencies, so like the count call it can be ran concurrently.